cypress/e2e/spec.cy.js

```diff
@@ -1,6 +1,58 @@
-// example that will be replaced by actual e2e tests
-describe('template spec', function() {
-    it('passes', () => {
-        assert(true)
-    })
-  })
\ No newline at end of file
+it('Chart is generated', function () {
+    cy.visit('/')
+    cy.findAllByText("Line").click()
+    cy.findAllByLabelText("Chart title").type('Cat super powers')
+    cy.findAllByLabelText("X label").type('cats')
+    cy.findAllByLabelText("Y label").type('powers')
+    cy.findAllByLabelText("X").type('1')
+    cy.findAllByLabelText("Y").type('3')
+    cy.findAllByText("Generate chart").click()
+    cy.get('#chart-img').should('exist')
+})
+
+it('Data is saved between tabs', function () {
+    cy.visit('/')
+    cy.findAllByText("Line").click()
+    cy.findAllByLabelText("Chart title").type('Cat super powers')
+    cy.findAllByLabelText("X label").type('cats')
+    cy.findAllByLabelText("Y label").type('powers')
+    cy.findAllByLabelText("X").type('6')
+    cy.findAllByLabelText("Y").type('9')
+    cy.findAllByLabelText("X").first().invoke('val').should('equal', '6')
+    cy.findAllByLabelText("Y").first().invoke('val').should('equal', '9')
+    cy.findAllByLabelText("X label").parent().parent().children().should('have.length', 5)
+
+})
+
+it('Chart is saved to the gallary', function () {
+    cy.visit('/')
+    cy.findAllByText("Line").click()
+    cy.findAllByLabelText("Chart title").type('Cat super powers')
+    cy.findAllByLabelText("X label").type('cats')
+    cy.findAllByLabelText("Y label").type('powers')
+    cy.findAllByLabelText("X").type('4')
+    cy.findAllByLabelText("Y").type('2')
+    cy.findAllByText("Generate chart").click()
+    cy.findAllByText("Save chart").click()
+    cy.findAllByText("Gallery").click()
+    cy.get('#gallery').children().should('have.length', 1)
+})
+
+it('Chart in gallery opens with correct values', function () {
+    cy.visit('/')
+    cy.findAllByText("Line").click()
+    cy.findAllByLabelText("Chart title").type('Cat super powers')
+    cy.findAllByLabelText("X label").type('cats')
+    cy.findAllByLabelText("Y label").type('powers')
+    cy.findAllByLabelText("X").type('12')
+    cy.findAllByLabelText("Y").type('21')
+    cy.findAllByText("Generate chart").click()
+    cy.findAllByText("Save chart").click()
+    cy.findAllByText("Gallery").click()
+    cy.get('#gallery').children().click()
+    cy.findAllByLabelText("X").first().invoke('val').should('equal', '12')
+    cy.findAllByLabelText("Y").first().invoke('val').should('equal', '21')
+    cy.findAllByLabelText("X label").parent().parent().children().should('have.length', 7)
```

Comment:
What does this line check?

Reply:
Line 55, I mean

Reply:
It checks that the correct number of data points are in the graph (should be one set, which makes it equal 7 children things)

Reply:
2 for the x and y, two for the x and y axis names, 2 for the new blank x and y added by the page when refreshed, and one for the table header